codegen/facelift/templates/IPCServiceAdapter.template.cpp

```diff
@@ -109,42 +100,69 @@ void {{className}}::appendDBUSIntrospectionData(QTextStream &s) const
 {
     Q_UNUSED(s);   // For empty interfaces
     {% for property in interface.properties %}
-    ::facelift::DBusSignatureHelper::addPropertySignature<ServiceType::PropertyType_{{property.name}}>(s, "{{property.name}}", {{ property.readonly | cppBool }});
+    {% if not property.type.is_model %}
```

Comment:
The existing approach based on DBusSignatureHelper ensures that all properties of a given type share the same code, which results in a more compact binary code. This is not the case with your change.

Reply:
The existing approach based on DBusSignatureHelper was **simply wrong**, go ahead and check.
So a change is needed to fix a defect!

Reply:
A change is needed indeed, but the approach can be kept.